### DISCUSSION TITLE
How to keep header and footer from being re-rendered on page transition

### DISCUSSION THREAD

**Discussion by JonCognioDigital** [vercel/next.js#20702] [Help]
Hi all,

I have a header and footer which are consistent across most pages of my site (but not all). I currently have a <Header /> and <Footer /> component which I import into every page and place as the very first and very last element in the page component.

The header and footer both contain data which is pulled from a CMS and because I'm using getStaticProps to statically render the pages this currently needs to be fetched on every page and injected into the header and footer like this... 

<header navItems={props.navItems} />

Now, because I'm using static generation and every rendered page needs the correct menu items in the footer when generated I don't believe I can just move these components to the _app.js or _document.js. Is there a solution to this? Is it possible to put the header and footer in every page without it flashing and re-rendering when changing pages. Or, is there a way to pass data from getStaticProps to a custom app or document so that a full rendered SSR page has the correct header and footer?

**Comment by yurimagagnatto** [comment]
You can use layouts: [https://nextjs.org/docs/basic-features/layouts](https://nextjs.org/docs/basic-features/layouts)
